File: 算法/微医超级码/topic1.js

// n 为放大倍数
function output (n) {
  let origin = `*********@*********
  
**@@@@**@@@**@@@@**

*@****@**@**@****@*

**@****@***@****@**

***@****@@@****@***

**@****@***@****@**

*@****@*****@****@*

**@@@@*******@@@@**

*******************`
  if (n === 1) return origin
  let result = ''
  let prev = ''
  for (let i = 0; i < origin.length; i++) {
    let scale = n
    if (origin[i] === '\n') {
      while (scale) {
        result += prev
        result += '\n'
        scale--
      }
      prev = ''
    } else {
      while (scale) {
        prev += origin[i]
        scale--
      }
    }
  }

  return  result
}

console.log(output(2))
// output(2)

File: 算法/微医超级码/topic1.py
def f(origin, n):
    if n == 1:
        return origin
    result = ''
    line_result = ''
    for i in range(len(origin)):
        scale = n
        if origin[i] == '\n':
            line_result += origin[i]
            line_result = line_result*n
            result +=line_result
            line_result = ''
        else:
          while scale:
            line_result += origin[i]
            scale -= 1
    return result
input_sample = input()
result = f(input_sample,2)
print(result)
